components/confirm_modal.jsx

```diff
@@ -1,80 +1,28 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
+// @flow
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import * as React from 'react';
 import {Modal} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
-export default class ConfirmModal extends React.Component {
-    static propTypes = {
-
-        /*
-         * Set to show modal
-         */
-        show: PropTypes.bool.isRequired,
-
-        /*
-         * Title to use for the modal
-         */
-        title: PropTypes.node,
-
-        /*
-         * Message to display in the body of the modal
-         */
-        message: PropTypes.node,
-
-        /*
-         * The CSS class to apply to the confirm button
-         */
-        confirmButtonClass: PropTypes.string,
-
-        /*
-         * The CSS class to apply to the modal
-         */
-        modalClass: PropTypes.string,
-
-        /*
-         * Text/jsx element on the confirm button
-         */
-        confirmButtonText: PropTypes.node,
-
-        /*
-         * Text/jsx element on the cancel button
-         */
-        cancelButtonText: PropTypes.node,
-
-        /*
-         * Set to show checkbox
-         */
-        showCheckbox: PropTypes.bool,
-
-        /*
-         * Text/jsx element to display with the checkbox
-         */
-        checkboxText: PropTypes.node,
-
-        /*
-         * Function called when the confirm button or ENTER is pressed. Passes `true` if the checkbox is checked
-         */
-        onConfirm: PropTypes.func.isRequired,
-
-        /*
-         * Function called when the cancel button is pressed or the modal is hidden. Passes `true` if the checkbox is checked
-         */
-        onCancel: PropTypes.func.isRequired,
-
-        /**
-         * Function called when modal is dismissed
-         */
-        onExited: PropTypes.func,
-
-        /*
-         * Set to hide the cancel button
-         */
-        hideCancel: PropTypes.bool,
-    }
-
+type Props = {|
```

Comment:
```suggestion
type Props = {
    +show: boolean,
    +title: React.Node,
    +message: React.Node,
    +confirmButtonClass: string,
    +modalClass: string,
    +confirmButtonText: React.Node,
    +cancelButtonText?: React.Node,
    +showCheckbox?: boolean,
    +checkboxText: React.Node,
    +onConfirm: (boolean) => void,
    +onCancel: (boolean) => void,
    +onExited: () => void,
    +hideCancel?: boolean,
}
```

`+` means read-only property, and `void` is the type for `void(0)`(`undefined`)